custom/icds_reports/static/js/directives/medicine-kit/medicine-kit.directive.js

```diff
@@ -1,228 +1,80 @@
-                        0 : parseInt(((min - range/10)/100).toFixed(2)),
-                    parseInt(((max + range/10)/100).toFixed(2)),
-                ];
-            }
-        });
-    };
-
-    vm.init = function() {
-        var locationId = vm.filtersData.location_id || vm.userLocationId;
-        if (!locationId || ["all", "null", "undefined"].indexOf(locationId) >= 0) {
-            vm.loadData();
-            vm.loaded = true;
-            return;
-        }
-        locationsService.getLocation(locationId).then(function(location) {
-            vm.location = location;
-            vm.loadData();
-            vm.loaded = true;
-        });
+        vm.setStepsMapLabel();
+        vm.myPromise = infrastructureService.getMedicineKitData(vm.step, vm.filtersData).then(
+            vm.loadDataFromResponse(true)
+        );
     };
 
     vm.init();
 
-    $scope.$on('filtersChange', function() {
-        vm.loadData();
-    });
-
-    vm.getDisableIndex = function () {
-        var i = -1;
-        window.angular.forEach(vm.selectedLocations, function (key, value) {
-            if (key !== null && key.location_id === vm.userLocationId) {
-                i = value;
-            }
-        });
-        return i;
-    };
-
-    vm.moveToLocation = function(loc, index) {
-        if (loc === 'national') {
-            $location.search('location_id', '');
-            $location.search('selectedLocationLevel', -1);
-            $location.search('location_name', '');
-        } else {
-            $location.search('location_id', loc.location_id);
-            $location.search('selectedLocationLevel', index);
-            $location.search('location_name', loc.name);
-        }
-    };
-
-    vm.chartOptions = {
-        chart: {
-            type: 'lineChart',
-            height: 450,
-            margin : {
-                top: 20,
-                right: 60,
-                bottom: 60,
-                left: 80,
-            },
-            x: function(d){ return d.x; },
-            y: function(d){ return d.y; },
-
-            color: d3.scale.category10().range(),
-            useInteractiveGuideline: true,
-            clipVoronoi: false,
-            tooltips: true,
-            xAxis: {
-                axisLabel: '',
-                showMaxMin: true,
-                tickFormat: function(d) {
-                    return d3.time.format('%b %Y')(new Date(d));
-                },
-                tickValues: function() {
-                    return vm.chartTicks;
-                },
-                axisLabelDistance: -100,
-            },
-
-            yAxis: {
-                axisLabel: '',
-                tickFormat: function(d){
-                    return d3.format(".2%")(d);
-                },
-                axisLabelDistance: 20,
-                forceY: [0],
+    vm.chartOptions = vm.getChartOptions(
+        '%b %Y', ".2%", ' Percentage of AWCs that reported having a Medicine Kit'
```

Comment:
I think this function should take as a parameter the object, something like:
```
vm.getChartOptions({
       xAxisFormat: '%b %Y',
       yAxisFormat: '.2%',
       title: 'Percentage of AWCs that reported having a Medicine Kit' 
})
```
Will be more readable.